### PR TITLE
Option 1: TPCDS test and benchmark for query 1 with all dsdgen files

### PR DESCRIPTION
**Pull request by isadikov**

### CODE REVIEW

**Comment by isadikov** [comment]
@aditi-pandit This is the PR with TPCDS code. You can run the test for query 1 in `velox/dwio/parquet/tests/ParquetTpcdsTest.cpp` similar to ParquetTpchTest.cpp.

The benchmark can be launched by running main method in `velox/benchmarks/tpcds/TpcdsBenchmark.cpp`. 

All of that is from CLion IDE.

I am still working on the comparison with DuckDB, the script https://www.notion.so/ahanaio/Velox-vs-DuckDb-TPC-H-Micro-benchmarks-summary-7798ed0075074870a17a72d830de069e#4cb60027e4ca4e44be6b3840a34a3fc9 requires a bit of work to get it running for TPCDS.

**Comment by majetideepak** [comment]
Is this work for Ahana only or is the plan to contribute to OSS? If it is later, we should open an issue and discuss the plan. We could build a TPCDS connector instead of importing so much code (dependency) from DuckDB.

**Comment by aditi-pandit** [comment]
> Is this work for Ahana only or is the plan to contribute to OSS? If it is later, we should open an issue and discuss the plan. We could build a TPCDS connector instead of importing so much code (dependency) from DuckDB.

@majetideepak : The plan is to contribute to OSS. Yes, we should open an issue and discuss this work at a high-level before submitting any code. I asked @isadikov to create this PR so that we have a prototype to use as a basis in the discussion.

We need to look into the code imported from DuckDB in this PR. It would improve by minimizing it.

I did investigate on the lines of TPC-DS connector.

- That code too would require the dsdgen import of this PR.

- The DuckdB dsdgen also needs a bit of refactoring to get to the point that we can do a TPC-DS connector.     
   In a nutshell, the TPC-H connector code had means to generate a subset of rows at a time, so it was easy to use it with the Velox connector/source/split API. TPC-DS does a single time data generation. So using it as is would mean a greater initialization time for the plugin. It also does not amortize for the data sources actually used in queries as such. 

In general, that is a much bigger undertaking and a big deviation from our effort of micro-benchmarking.

Lets review and iterate on this PR a bit I feel till we reach a point to share ideas more broadly with the community.

**Comment by isadikov** [comment]
I checked the code and all of the imported files are required to generate the TPC-DS dataset. I could prepare a binary that we can check in or a static library to minimise the code if you prefer this approach instead.

**Comment by majetideepak** [comment]
Let's discuss the options in a Velox OSS issue.

**Comment by isadikov** [comment]
Presto plan with decimals: 

```
- Output[c_customer_id] => [c_customer_id:char(16)]
    - TopN[100 by (c_customer_id ASC_NULLS_LAST)] => [c_customer_id:char(16)]
        - LocalExchange[SINGLE] () => [c_customer_id:char(16)]
            - RemoteStreamingExchange[GATHER] => [c_customer_id:char(16)]
                - TopNPartial[100 by (c_customer_id ASC_NULLS_LAST)] => [c_customer_id:char(16)]
                    - FilterProject[filterPredicate = (CAST(sum AS double)) > ((CAST(COALESCE(expr_86, avg_97) AS double)) * (DOUBLE''1.2'')), projectLocality = LOCAL] => [c_customer_id:char(16)]
                        - CrossJoin => [sum:decimal(38,2), c_customer_id:char(16), expr_86:decimal(38,2), avg_97:decimal(38,2)]
                                Distribution: REPLICATED
                            - LeftJoin[("sr_store_sk" = "sr_store_sk_24")] => [sum:decimal(38,2), c_customer_id:char(16), expr_86:decimal(38,2)]
                                    Distribution: PARTITIONED
                                - RemoteStreamingExchange[REPARTITION] => [c_customer_id:char(16), sr_store_sk:bigint, sum:decimal(38,2)]
                                    - InnerJoin[("c_customer_sk" = "sr_customer_sk")] => [c_customer_id:char(16), sr_store_sk:bigint, sum:decimal(38,2)]
                                            Distribution: PARTITIONED
                                        - RemoteStreamingExchange[REPARTITION] => [c_customer_sk:bigint, c_customer_id:char(16)]
                                                Estimates: {rows: 100000 (2.00MB), cpu: 6000000.00, memory: 0.00, network: 3000000.00}
                                            - TableScan[TableHandle {connectorId=''tpcds'', connectorHandle=''tpcds:customer:sf1.0'', layout=''Optional[customer]''}] => [c_customer_sk:bigint, c_customer_id:char(16)]
                                                    Estimates: {rows: 100000 (2.00MB), cpu: 3000000.00, memory: 0.00, network: 0.00}
                                                    LAYOUT: customer
                                                    c_customer_sk := tpcds:c_customer_sk (1:334)
                                                    c_customer_id := tpcds:c_customer_id (1:334)
                                        - LocalExchange[HASH] (sr_customer_sk) => [sr_customer_sk:bigint, sr_store_sk:bigint, sum:decimal(38,2)]
                                            - RemoteStreamingExchange[REPARTITION] => [sr_customer_sk:bigint, sr_store_sk:bigint, sum:decimal(38,2)]
                                                - InnerJoin[("sr_store_sk" = "s_store_sk")] => [sr_customer_sk:bigint, sr_store_sk:bigint, sum:decimal(38,2)]
                                                        Distribution: PARTITIONED
                                                    - RemoteStreamingExchange[REPARTITION] => [sr_customer_sk:bigint, sr_store_sk:bigint, sum:decimal(38,2)]
                                                        - Aggregate(FINAL)[sr_customer_sk, sr_store_sk] => [sr_customer_sk:bigint, sr_store_sk:bigint, sum:decimal(38,2)]
                                                                sum := "presto.default.sum"((sum_98)) (1:111)
                                                            - LocalExchange[HASH] (sr_customer_sk, sr_store_sk) => [sr_customer_sk:bigint, sr_store_sk:bigint, sum_98:varbinary]
                                                                - RemoteStreamingExchange[REPARTITION] => [sr_customer_sk:bigint, sr_store_sk:bigint, sum_98:varbinary]
                                                                    - Aggregate(PARTIAL)[sr_customer_sk, sr_store_sk] => [sr_customer_sk:bigint, sr_store_sk:bigint, sum_98:varbinary]
                                                                            sum_98 := "presto.default.sum"((sr_return_amt)) (1:111)
                                                                        - InnerJoin[("sr_returned_date_sk" = "d_date_sk")] => [sr_customer_sk:bigint, sr_store_sk:bigint, sr_return_amt:decimal(7,2)]
                                                                                Estimates: {rows: 50350 (2.64MB), cpu: 33478460.20, memory: 3270.85, network: 10032830.85}
                                                                                Distribution: PARTITIONED
                                                                            - RemoteStreamingExchange[REPARTITION] => [sr_returned_date_sk:bigint, sr_customer_sk:bigint, sr_store_sk:bigint, sr_return_amt:decimal(7,2)]
                                                                                    Estimates: {rows: 287514 (15.08MB), cpu: 20059120.00, memory: 0.00, network: 10029560.00}
                                                                                - TableScan[TableHandle {connectorId=''tpcds'', connectorHandle=''tpcds:store_returns:sf1.0'', layout=''Optional[store_returns]''}] => [sr_returned_date_sk:bigint, sr_customer_sk:bigint, sr_store_sk:bigint, sr_return_amt:decimal(7,2)]
                                                                                        Estimates: {rows: 287514 (15.08MB), cpu: 10029560.00, memory: 0.00, network: 0.00}
                                                                                        LAYOUT: store_returns
                                                                                        sr_returned_date_sk := tpcds:sr_returned_date_sk (1:155)
                                                                                        sr_store_sk := tpcds:sr_store_sk (1:155)
                                                                                        sr_customer_sk := tpcds:sr_customer_sk (1:155)
                                                                                        sr_return_amt := tpcds:sr_return_amt (1:155)
                                                                            - LocalExchange[HASH] (d_date_sk) => [d_date_sk:bigint]
                                                                                    Estimates: {rows: 363 (19.52kB), cpu: 2055184.55, memory: 0.00, network: 3270.85}
                                                                                - RemoteStreamingExchange[REPARTITION] => [d_date_sk:bigint]
                                                                                        Estimates: {rows: 363 (19.52kB), cpu: 2051913.70, memory: 0.00, network: 3270.85}
                                                                                    - ScanFilterProject[table = TableHandle {connectorId=''tpcds'', connectorHandle=''tpcds:date_dim:sf1.0'', layout=''Optional[date_dim]''}, filterPredicate = (d_year) = (INTEGER''2000''), projectLocality = LOCAL] => [d_date_sk:bigint]
                                                                                            Estimates: {rows: 73049 (3.83MB), cpu: 1022686.00, memory: 0.00, network: 0.00}/{rows: 363 (19.52kB), cpu: 2045372.00, memory: 0.00, network: 0.00}/{rows: 363 (19.52kB), cpu: 2048642.85, memory: 0.00, network: 0.00}
                                                                                            LAYOUT: date_dim
                                                                                            d_date_sk := tpcds:d_date_sk (1:170)
                                                                                            d_year := tpcds:d_year (1:170)
                                                    - LocalExchange[HASH] (s_store_sk) => [s_store_sk:bigint]
                                                            Estimates: {rows: 12 (660B), cpu: 556.00, memory: 0.00, network: 108.00}
                                                        - RemoteStreamingExchange[REPARTITION] => [s_store_sk:bigint]
                                                                Estimates: {rows: 12 (660B), cpu: 448.00, memory: 0.00, network: 108.00}
                                                            - ScanFilter[table = TableHandle {connectorId=''tpcds'', connectorHandle=''tpcds:store:sf1.0'', layout=''Optional[store]''}, filterPredicate = (s_state) = (CHAR''TN'')] => [s_store_sk:bigint, s_state:char(2)]
                                                                    Estimates: {rows: 12 (660B), cpu: 170.00, memory: 0.00, network: 0.00}/{rows: 12 (660B), cpu: 340.00, memory: 0.00, network: 0.00}
                                                                    LAYOUT: store
                                                                    s_state := tpcds:s_state (1:327)
                                                                    s_store_sk := tpcds:s_store_sk (1:327)
                                - Aggregate(FINAL)[sr_store_sk_24] => [sr_store_sk_24:bigint, expr_86:decimal(38,2)]
                                        expr_86 := "presto.default.avg"((avg_99)) (1:381)
                                    - LocalExchange[HASH] (sr_store_sk_24) => [sr_store_sk_24:bigint, avg_99:varbinary]
                                        - RemoteStreamingExchange[REPARTITION] => [sr_store_sk_24:bigint, avg_99:varbinary]
                                            - Aggregate(PARTIAL)[sr_store_sk_24] => [sr_store_sk_24:bigint, avg_99:varbinary]
                                                    avg_99 := "presto.default.avg"((sum_70)) (1:381)
                                                - Project[projectLocality = LOCAL] => [sr_store_sk_24:bigint, sum_70:decimal(38,2)]
                                                    - Aggregate(FINAL)[sr_customer_sk_20, sr_store_sk_24] => [sr_customer_sk_20:bigint, sr_store_sk_24:bigint, sum_70:decimal(38,2)]
                                                            sum_70 := "presto.default.sum"((sum_100)) (1:111)
                                                        - LocalExchange[HASH] (sr_customer_sk_20, sr_store_sk_24) => [sr_customer_sk_20:bigint, sr_store_sk_24:bigint, sum_100:varbinary]
                                                            - RemoteStreamingExchange[REPARTITION] => [sr_customer_sk_20:bigint, sr_store_sk_24:bigint, sum_100:varbinary]
                                                                - Aggregate(PARTIAL)[sr_customer_sk_20, sr_store_sk_24] => [sr_customer_sk_20:bigint, sr_store_sk_24:bigint, sum_100:varbinary]
                                                                        sum_100 := "presto.default.sum"((sr_return_amt_28)) (1:111)
                                                                    - InnerJoin[("sr_returned_date_sk_17" = "d_date_sk_37")] => [sr_customer_sk_20:bigint, sr_store_sk_24:bigint, sr_return_amt_28:decimal(7,2)]
                                                                            Estimates: {rows: 50350 (2.64MB), cpu: 33478460.20, memory: 3270.85, network: 10032830.85}
                                                                            Distribution: PARTITIONED
                                                                        - RemoteStreamingExchange[REPARTITION] => [sr_returned_date_sk_17:bigint, sr_customer_sk_20:bigint, sr_store_sk_24:bigint, sr_return_amt_28:decimal(7,2)]
                                                                                Estimates: {rows: 287514 (15.08MB), cpu: 20059120.00, memory: 0.00, network: 10029560.00}
                                                                            - TableScan[TableHandle {connectorId=''tpcds'', connectorHandle=''tpcds:store_returns:sf1.0'', layout=''Optional[store_returns]''}] => [sr_returned_date_sk_17:bigint, sr_customer_sk_20:bigint, sr_store_sk_24:bigint, sr_return_amt_28:decimal(7,2)]
                                                                                    Estimates: {rows: 287514 (15.08MB), cpu: 10029560.00, memory: 0.00, network: 0.00}
                                                                                    LAYOUT: store_returns
                                                                                    sr_customer_sk_20 := tpcds:sr_customer_sk (1:155)
                                                                                    sr_return_amt_28 := tpcds:sr_return_amt (1:155)
                                                                                    sr_store_sk_24 := tpcds:sr_store_sk (1:155)
                                                                                    sr_returned_date_sk_17 := tpcds:sr_returned_date_sk (1:155)
                                                                        - LocalExchange[HASH] (d_date_sk_37) => [d_date_sk_37:bigint]
                                                                                Estimates: {rows: 363 (19.52kB), cpu: 2055184.55, memory: 0.00, network: 3270.85}
                                                                            - RemoteStreamingExchange[REPARTITION] => [d_date_sk_37:bigint]
                                                                                    Estimates: {rows: 363 (19.52kB), cpu: 2051913.70, memory: 0.00, network: 3270.85}
                                                                                - ScanFilterProject[table = TableHandle {connectorId=''tpcds'', connectorHandle=''tpcds:date_dim:sf1.0'', layout=''Optional[date_dim]''}, filterPredicate = (d_year_43) = (INTEGER''2000''), projectLocality = LOCAL] => [d_date_sk_37:bigint]
                                                                                        Estimates: {rows: 73049 (3.83MB), cpu: 1022686.00, memory: 0.00, network: 0.00}/{rows: 363 (19.52kB), cpu: 2045372.00, memory: 0.00, network: 0.00}/{rows: 363 (19.52kB), cpu: 2048642.85, memory: 0.00, network: 0.00}
                                                                                        LAYOUT: date_dim
                                                                                        d_year_43 := tpcds:d_year (1:170)
                                                                                        d_date_sk_37 := tpcds:d_date_sk (1:170)
                            - LocalExchange[SINGLE] () => [avg_97:decimal(38,2)]
                                    Estimates: {rows: 1 (55B), cpu: 1.00, memory: 17.00, network: 17.00}
                                - RemoteStreamingExchange[REPLICATE] => [avg_97:decimal(38,2)]
                                        Estimates: {rows: 1 (55B), cpu: 1.00, memory: 17.00, network: 17.00}
                                    - Aggregate => [avg_97:decimal(38,2)]
                                            Estimates: {rows: 1 (55B), cpu: 1.00, memory: 17.00, network: 0.00}
                                            avg_97 := "presto.default.avg"((expr_95)) (1:381)
                                        - Values => [expr_95:decimal(38,2)]
                                                Estimates: {rows: 1 (55B), cpu: 0.00, memory: 0.00, network: 0.00}
                                                (null)

```

Velox plan with decimals:

If I use decimals exclusively, I get this:
> Reason: Scalar function signature is not supported: gt(DECIMAL(38,2), DECIMAL(38,4)). Supported signatures: (date,date) -> boolean, (timestamp,timestamp) -> boolean, (long_decimal(a_precision,a_scale),long_decimal(a_precision,a_scale)) -> boolean, (boolean,boolean) -> boolean, (short_decimal(a_precision,a_scale),short_decimal(a_precision,a_scale)) -> boolean, (varbinary,varbinary) -> boolean, (varchar,varchar) -> boolean, (tinyint,tinyint) -> boolean, (smallint,smallint) -> boolean, (integer,integer) -> boolean, (bigint,bigint) -> boolean, (real,real) -> boolean, (double,double) -> boolean.

If I try to cast decimals to double when comparing like this: `cast(ctr_total_return as double) > 1.2 * cast(avg_ctr_total_return as double)`, the query returns empty results due to cast decimal -> double, it is actually the same the other way around, I tried to explain it here: https://ahana-io.slack.com/archives/C02AK36GN22/p1676269385833989?thread_ts=1675980898.289549&cid=C02AK36GN22.

Plan that produces an empty result due to decimal->double cast (the plan is similar to the Presto plan above):
```
-- TopN[100 c_customer_id ASC NULLS LAST] -> c_customer_id:VARCHAR
  -- HashJoin[INNER ctr_customer_sk=c_customer_sk] -> c_customer_id:VARCHAR
    -- HashJoin[INNER ctr_store_sk=ctr_store_sk, filter: gt(cast ROW["ctr_total_return"] as DOUBLE,multiply(1.2,cast ROW["avg_ctr_total_return"] as DOUBLE))] -> ctr_customer_sk:INTEGER
      -- HashJoin[INNER ctr_store_sk=s_store_sk] -> ctr_store_sk:INTEGER, ctr_total_return:DECIMAL(38,2), ctr_customer_sk:INTEGER
        -- Project[expressions: (ctr_customer_sk:INTEGER, ROW["sr_customer_sk"]), (ctr_store_sk:INTEGER, ROW["sr_store_sk"]), (ctr_total_return:DECIMAL(38,2), ROW["ctr_total_return"])] -> ctr_customer_sk:INTEGER, ctr_store_sk:INTEGER, ctr_total_return:DECIMAL(38,2)
          -- Aggregation[FINAL [sr_customer_sk, sr_store_sk] ctr_total_return := sum("ctr_total_return")] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DECIMAL(38,2)
            -- LocalPartition[REPARTITION] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DECIMAL(38,2)
              -- Aggregation[PARTIAL [sr_customer_sk, sr_store_sk] ctr_total_return := sum(ROW["sr_return_amt"])] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DECIMAL(38,2)
                -- HashJoin[INNER sr_returned_date_sk=d_date_sk] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, sr_return_amt:DECIMAL(7,2)
                  -- TableScan[table: store_returns] -> sr_returned_date_sk:INTEGER, sr_customer_sk:INTEGER, sr_return_amt:DECIMAL(7,2), sr_store_sk:INTEGER
                  -- TableScan[table: date_dim, range filters: [(d_year, BigintRange: [2000, 2000] no nulls)]] -> d_date_sk:INTEGER, d_year:INTEGER
        -- TableScan[table: store, range filters: [(s_state, Filter(BytesValues, deterministic, null not allowed))]] -> s_store_sk:INTEGER, s_state:VARCHAR
      -- Project[expressions: (ctr_store_sk:INTEGER, ROW["sr_store_sk"]), (avg_ctr_total_return:DECIMAL(38,2), ROW["avg_ctr_total_return"])] -> ctr_store_sk:INTEGER, avg_ctr_total_return:DECIMAL(38,2)
        -- Aggregation[FINAL [sr_store_sk] avg_ctr_total_return := avg("avg_ctr_total_return")] -> sr_store_sk:INTEGER, avg_ctr_total_return:DECIMAL(38,2)
          -- LocalPartition[REPARTITION] -> sr_store_sk:INTEGER, avg_ctr_total_return:VARBINARY
            -- Aggregation[PARTIAL [sr_store_sk] avg_ctr_total_return := avg(ROW["ctr_total_return"])] -> sr_store_sk:INTEGER, avg_ctr_total_return:VARBINARY
              -- Aggregation[INTERMEDIATE [sr_customer_sk, sr_store_sk] ctr_total_return := sum("ctr_total_return")] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DECIMAL(38,2)
                -- Aggregation[PARTIAL [sr_customer_sk, sr_store_sk] ctr_total_return := sum(ROW["sr_return_amt"])] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DECIMAL(38,2)
                  -- HashJoin[INNER sr_returned_date_sk=d_date_sk] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, sr_return_amt:DECIMAL(7,2)
                    -- TableScan[table: store_returns] -> sr_returned_date_sk:INTEGER, sr_customer_sk:INTEGER, sr_return_amt:DECIMAL(7,2), sr_store_sk:INTEGER
                    -- TableScan[table: date_dim, range filters: [(d_year, BigintRange: [2000, 2000] no nulls)]] -> d_date_sk:INTEGER, d_year:INTEGER
    -- TableScan[table: customer] -> c_customer_sk:INTEGER, c_customer_id:VARCHAR
```

That is why I had to keep doubles in the TPC-DS dataset (similar to TPC-H), this way query works and produces correct results.

Plan of a working Velox query producing correct results:
```
-- TopN[100 c_customer_id ASC NULLS LAST] -> c_customer_id:VARCHAR
  -- HashJoin[INNER ctr_customer_sk=c_customer_sk] -> c_customer_id:VARCHAR
    -- HashJoin[INNER ctr_store_sk=ctr_store_sk, filter: gt(ROW["ctr_total_return"],ROW["avg_ctr_total_return"])] -> ctr_customer_sk:INTEGER
      -- HashJoin[INNER ctr_store_sk=s_store_sk] -> ctr_store_sk:INTEGER, ctr_total_return:DOUBLE, ctr_customer_sk:INTEGER
        -- Project[expressions: (ctr_customer_sk:INTEGER, ROW["sr_customer_sk"]), (ctr_store_sk:INTEGER, ROW["sr_store_sk"]), (ctr_total_return:DOUBLE, ROW["ctr_total_return"])] -> ctr_customer_sk:INTEGER, ctr_store_sk:INTEGER, ctr_total_return:DOUBLE
          -- Aggregation[FINAL [sr_customer_sk, sr_store_sk] ctr_total_return := sum("ctr_total_return")] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DOUBLE
            -- LocalPartition[REPARTITION] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DOUBLE
              -- Aggregation[PARTIAL [sr_customer_sk, sr_store_sk] ctr_total_return := sum(ROW["sr_return_amt"])] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DOUBLE
                -- HashJoin[INNER sr_returned_date_sk=d_date_sk] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, sr_return_amt:DOUBLE
                  -- TableScan[table: store_returns] -> sr_returned_date_sk:INTEGER, sr_customer_sk:INTEGER, sr_return_amt:DOUBLE, sr_store_sk:INTEGER
                  -- TableScan[table: date_dim, range filters: [(d_year, BigintRange: [2000, 2000] no nulls)]] -> d_date_sk:INTEGER, d_year:INTEGER
        -- TableScan[table: store, range filters: [(s_state, Filter(BytesValues, deterministic, null not allowed))]] -> s_store_sk:INTEGER, s_state:VARCHAR
      -- Project[expressions: (ctr_store_sk:INTEGER, ROW["sr_store_sk"]), (avg_ctr_total_return:DOUBLE, multiply(ROW["avg_ctr_total_return"],1.2))] -> ctr_store_sk:INTEGER, avg_ctr_total_return:DOUBLE
        -- Aggregation[FINAL [sr_store_sk] avg_ctr_total_return := avg("avg_ctr_total_return")] -> sr_store_sk:INTEGER, avg_ctr_total_return:DOUBLE
          -- LocalPartition[REPARTITION] -> sr_store_sk:INTEGER, avg_ctr_total_return:ROW<"":DOUBLE,"":BIGINT>
            -- Aggregation[PARTIAL [sr_store_sk] avg_ctr_total_return := avg(ROW["ctr_total_return"])] -> sr_store_sk:INTEGER, avg_ctr_total_return:ROW<"":DOUBLE,"":BIGINT>
              -- Aggregation[INTERMEDIATE [sr_customer_sk, sr_store_sk] ctr_total_return := sum("ctr_total_return")] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DOUBLE
                -- Aggregation[PARTIAL [sr_customer_sk, sr_store_sk] ctr_total_return := sum(ROW["sr_return_amt"])] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, ctr_total_return:DOUBLE
                  -- HashJoin[INNER sr_returned_date_sk=d_date_sk] -> sr_customer_sk:INTEGER, sr_store_sk:INTEGER, sr_return_amt:DOUBLE
                    -- TableScan[table: store_returns] -> sr_returned_date_sk:INTEGER, sr_customer_sk:INTEGER, sr_return_amt:DOUBLE, sr_store_sk:INTEGER
                    -- TableScan[table: date_dim, range filters: [(d_year, BigintRange: [2000, 2000] no nulls)]] -> d_date_sk:INTEGER, d_year:INTEGER
    -- TableScan[table: customer] -> c_customer_sk:INTEGER, c_customer_id:VARCHAR
```
